[FE] #86 질문 상세 조회 페이지 통신 로직 구현 (GET, POST 메서드)

novice1993 opened the pull request:
- 질문 상세 조회 페이지 서버 통신 로직 구현 (질문, 답변, 댓글 조회 / 답변, 댓글 추가)
- 이외 변경사항 : 서버 데이터 적용에 따른 컴포넌트 구조 변경 및 데이터 적용에 따른 CSS 일부 수정

novice1993 committed:
:sparkles: Feat : post 메서드 기본틀 구현

- 질문 상세 조회 페이지의 답변 작성 관련 post 메서드 기본틀 구현
- JSON-Server 라이브러리로 테스트 완료하였으며, BE에서 API 제공시 전송한 데이터 형태 수정할 예정

Issues #50

novice1993 committed:
Merge branch 'dev-fe' of github.com:codestates-seb/seb45_pre_017 into dev-fe#50/httpBasicLogic-postTest

novice1993 committed:
Merge branch 'dev-fe' of github.com:codestates-seb/seb45_pre_017 into dev-fe#50/httpBasicLogic-postTest

novice1993 committed:
:sparkles: 서버 통신 로직 구현 (GET, POST 메서드)

- 질문 상세 조회 페이지 서버 통신 로직 구현 (질문, 답변, 댓글 조회 / 답변, 댓글 추가)
- 이외 변경사항 : 서버 데이터 적용에 따른 컴포넌트 구조 변경 및 데이터 적용에 따른 CSS 일부 수정

Issues #86

novice1993 committed:
:twisted_rightwards_arrows: Feat : 브랜치 병합으로 인한 커밋

- dev-fe 브랜치 병합으로 인한 커밋

Issues #86

kimmand0o0 commented on client/data/data.json:
```diff
@@ -29,6 +29,21 @@
+    },
+    {
+      "value": "<p>123123123</p><p>12412412412</p>",
+      "id": 6
```
어떤 테스트를 위한 데이터인지 명시해주시면 좋을 것 같습니다!

novice1993 replied:
이 부분은 서버 구현 전 테스트를 위한 json-server 라이브러리 부분이라 삭제해도 괜찮을 거 같습니다 'ㅡ'

kimmand0o0 commented on client/src/App.tsx:
```diff
@@ -21,7 +21,7 @@ function App() {
           <Routes>
             <Route path="/" element={<Mainpage />} />
-            <Route path="/page/:votes" element={<QuestionContentPage />} />
+            <Route path="/posts/1" element={<QuestionContentPage />} />
```
path에서 변수를 사용하지 않고 1을 사용하신 이유가 있을까요?

novice1993 replied:
예전에 수정한 데이터라 정확히 기억이 나지는 않는데 특별한 이유가 있는 건 아니고,
현재 Main 페이지 구현이 안된 상황이라 사실상 path에 어떤 파라미터를 주든 동일한 상황이기 때문에 적당한 값을 설정한 것으로 보입니다 'ㅡ'
( 현재 path를 통해 전달된 파라미터를 사용해서 개별 질문 데이터를 조회하는 것이 아니기 때문에 )

kimmand0o0 replied:
패스가 1로 고정되면 질문 작성, 수정, 답변 수정 페이지에서 리다이렉트 되는 페이지가 보이지 않을 것 같습니다.
혹시 수정이 많이 어려울까요?

novice1993 replied:
메인 페이지 구현 후 
:posts 형식으로 파라미터를 넘길 수만 있다면 수정은 간단할 것 같습니다

kimmand0o0 replied:
넵 확인입니다!

kimmand0o0 commented on client/src/components/AnswerComment/Comment.tsx:
```diff
@@ -1,29 +1,32 @@
-  const { text, writer, date } = props;
+const Comment = () => {
+  const userID: string = "1";
+  const answerID: string = "21";
```
두 Id를 number 가 아닌 string으로 사용하신 이유가 있을까요?

novice1993 replied:
Main 페이지가 구현된다면 해당 파라미터를 메인 페이지에서 전달받아서 사용하게 될텐데
파라미터로 전달받을 경우 데이터 타입이 number가 아니라 string이기 때문에 string으로 설정하였습니다 'ㅡ'

kimmand0o0 replied:
넵 확인입니다!

udaeng8286 reviewed (commented):
확인 완료했습니다!

novice1993 committed:
:bug: Fix : 답변/댓글 GET 관련 예외처리 추가

- Comment 컴넌트에 answerID props를 전달해서 답변에 해당하는 댓글이 조회되도록 수정 (기존에는 answerID를 하나의 값으로 고정해서 테스트)
- 질문에 대한 답변, 답변에 대한 댓글 없을 시 서버에서 값이 undefined로 반환되어 발생하던 오류 수정 (undefined 일 경우 null 값을 반환하여 화면에 아무런 표시 안되도록 설정)

Issues #86

novice1993 committed:
:fire: Fix : 불필요한 파일 삭제

- 서버 구현 전 테스트를 위한 JSON-Server 라이브러리 관련 파일 삭제

Issues #86

novice1993 commented:
@udaeng8286 @kimmand0o0 

답변/댓글 관련 예외처리 추가 및 불필요한 파일 삭제 (JSON-Server 라이브러리 관련 파일) 후 커밋 추가해서 리뷰 한번만 더 부탁드립니다 'ㅡ'!!!

kimmand0o0 reviewed (approved):
확인 완료입니다! 고생하셨습니다!

udaeng8286 reviewed (commented):
확인 완료했습니다! 고생 많으셨습니다

novice1993 committed:
:twisted_rightwards_arrows: Feat : 브랜치 병합

- conflict 해결위해 dev-fe 브랜치와 병합

Issues #86
